pkg/sql/alter_table.go

```diff
@@ -906,11 +905,73 @@ func applyColumnMutation(
 				col.Type.SQLString(), typ.SQLString())
 		case schemachange.ColumnConversionTrivial:
 			col.Type = *typ
-		default:
-			return unimplemented.NewWithIssueDetail(9851,
-				fmt.Sprintf("%s->%s", col.Type.SQLString(), typ.SQLString()),
-				"type conversion not yet implemented")
-		}
+
+		default: // try conversion
+			//1. Create a new computed column c' using c::t'
+			// Computed off old column, need to update ComputeExpr
+			// Make shadow column for now
+
+			// Can concurrent schema changes be happening?
+			currentMutationID := tableDesc.ClusterVersion.NextMutationID
+			for i := range tableDesc.Mutations {
+				mut := &tableDesc.Mutations[i]
+				if mut.MutationID < currentMutationID {
+					return unimplemented.Newf(
+						"column type change while undergoing schema change",
+						"table %s is currently undergoing a schema change", tableDesc.Name)
+				}
+			}
+
+			nameExists := func(name string) bool {
+				_, _, err := tableDesc.FindColumnByName(tree.Name(name))
+				return err == nil
+			}
+
+			name := sqlbase.GenerateUniqueConstraintName(
+				col.Name,
+				nameExists,
+			)
+
+			var newColComputeExpr = proto.String(
+				fmt.Sprintf("%s::%s", col.Name, t.ToType.String()))
+			hasDefault := col.HasDefault()
+			var newColDefaultExpr *string
+			if hasDefault {
+				// TODO(richardjcai) Handle NULL
+				newColDefaultExpr = proto.String(
+					fmt.Sprintf("%s::%s", col.DefaultExprStr(), t.ToType.String()))
+			}
+
+			id := tableDesc.NextColumnID
+			tableDesc.NextColumnID++
+
+			newCol := sqlbase.ColumnDescriptor{
+				Name: name,
+				ID: id,
+				Type:*t.ToType,
+				Nullable: col.Nullable,
+				DefaultExpr: newColDefaultExpr,
+				Hidden: true,
+				UsesSequenceIds: col.UsesSequenceIds,
```

Comment:
This seems suspicious -- check the checks that drop column makes before deleting a column. That will tell you what sort of stuff a column depends on so we can go and update all of those references.